yt/frontends/cholla/data_structures.py

```diff
@@ -0,0 +1,145 @@
+        for i in range(self.num_grids):
+            g = self.grid(i, self, self.grid_levels.flat[i], self.grid_dimensions[i])
+            g._prepare_grid()
+            g._setup_dx()
+            self.grids[i] = g
+
+
+class ChollaDataset(Dataset):
+    _index_class = ChollaHierarchy
+    _field_info_class = ChollaFieldInfo
+
+    def __init__(
+        self,
+        filename,
+        dataset_type="cholla",
+        storage_filename=None,
+        units_override=None,
+    ):
+        self.fluid_types += ("cholla",)
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.storage_filename = storage_filename
+
+    def _set_code_unit_attributes(self):
+        # This is where quantities are created that represent the various
+        # on-disk units.  These are the currently available quantities which
+        # should be set, along with examples of how to set them to standard
+        # values.
+        #
+        self.length_unit = self.quan(1.0, "pc")
+        self.mass_unit = self.quan(1.0, "Msun")
+        self.time_unit = self.quan(1000, "yr")
+        self.velocity_unit = self.quan(1.0, "cm/s")
+        self.magnetic_unit = self.quan(1.0, "gauss")
+
+        # this minimalistic implementation fills the requirements for
+        # this frontend to run, change it to make it run _correctly_ !
+        for key, unit in self.__class__.default_units.items():
+            setdefaultattr(self, key, self.quan(1, unit))
+
+    def _parse_parameter_file(self):
+
+        h5f = h5py.File(self.parameter_filename, mode="r")
+        attrs = h5f.attrs
+        self.parameters = {k: v for (k, v) in attrs.items()}
+        self.domain_left_edge = attrs["bounds"][:].astype("=f8")
+        self.domain_right_edge = attrs["domain"][:].astype("=f8")
+        self.dimensionality = len(attrs["dims"][:])
+        self.domain_dimensions = attrs["dims"][:].astype("=f8")
+        self.current_time = attrs["t"][:]
+        self._periodicity = (False, False, False)
+        h5f.close()
+
+        # CHOLLA cannot yet be run as a cosmological simulation
+        self.cosmological_simulation = 0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+        # CHOLLA datasets are always unigrid cartesian
+        self.geometry = "cartesian"
+
+    @classmethod
+    def _is_valid(cls, filename, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+        #
+        # The functionality in this method should be unique enough that it can
+        # differentiate the frontend from others. Sometimes this means looking
+        # for specific fields or attributes in the dataset in addition to
+        # looking at the file name or extension.
+        try:
+            fileh = h5py.File(filename, mode="r")
+        except OSError:
```

Comment:
Not catching ImportError breaks any install where h5py isn't installed
```suggestion
        except (ImportError, OSError):
```